helix-core/src/main/java/org/apache/helix/manager/zk/ZKUtil.java

```diff
@@ -92,15 +97,17 @@ public static boolean isClusterSetup(String clusterName, HelixZkClient zkClient)
     requiredPaths.add(PropertyPathBuilder.controllerHistory(clusterName));
     boolean isValid = true;
 
-    BaseDataAccessor<Object> baseAccessor = new ZkBaseDataAccessor<Object>(zkClient);
-    boolean[] ret = baseAccessor.exists(requiredPaths, 0);
+    boolean[] ret = new boolean[requiredPaths.size()];
+    for (int i = 0; i < requiredPaths.size(); i++) {
+      ret[i] = zkClient.exists(requiredPaths.get(i));
```

Comment:
It seems this changes the original behavior. `baseAccessor.exits` is actually an async call using `asyncExists()` in the bottom implementation. I think it is better to keep the original behavior.

Reply:
This is a util function. Functionality wise, there is no difference.

Doing an asyncExists() would require more change (you have to look at the callbacks).

Reply:
I did that is why I know this is an async. If you do this sync, it changes the original behavior. If one call of exists fails and throw exception this call fails. But the original behavior won’t fail all the calls. Some other calls are still possible to succeed. Like you said this is just a realm aware change, I don’t think it is a good idea to change the behavior.

Reply:
Okay. Wrapping it with a try-catch.

Reply:
I am still not very convinced we change the behavior. I understand the functionality may be the same. But if one call is blocked because of connection and it has to retry, other calls could be blocked. I am not sure how critical it is for here. Maybe Junkai can give an opinion. Since we are also integrating baseDataAccessor, can we just keep this original behavior by calling base data accessor?

Reply:
It is not that critical. This could have easily used a set of sync exists call too.